maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java

```diff
@@ -186,6 +200,21 @@ private DistroProperties resolveDistroProperties(Server server) throws MojoExecu
 					}
 					platformMode = false;
 					break;
+				case O3_REFERENCE_APPLICATION:
+					String distro = generateO3ReferenceDistro();
+					if (StringUtils.isNotBlank(distro)) {
+						File distroFile = new File(server.getServerDirectory(), "distro.properties");
+						try {
+							FileWriter writer = new FileWriter(distroFile);
+							writer.write(distro);
+							writer.close();
```

Comment:
Will you be able to close this if something went wrong before reaching this line?